vm/systemSmartContracts/staking.go

```diff
@@ -629,8 +649,12 @@ func (r *stakingSC) unBond(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 		r.eei.AddReturnMessage(fmt.Sprintf("unBond is not possible for key %s which is staked", encodedBlsKey))
 		return vmcommon.UserError
 	}
-	if registrationData.Jailed {
-		r.eei.AddReturnMessage(fmt.Sprintf("unBond is not possible for jailed key %s", encodedBlsKey))
+	if !r.isActionAllowed(registrationData, args.Arguments[0]) {
+		r.eei.AddReturnMessage("cannot stake node which is jailed or with bad rating " + encodedBlsKey)
+		return vmcommon.UserError
+	}
+	if registrationData.Waiting {
```

Comment:
|| r.isInWaiting(args.Arguments[0])  ?

Reply:
No need for it. If it is with status waiting it is in waiting list

Reply:
ok